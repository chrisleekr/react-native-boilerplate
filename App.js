import React from 'react';
import {StyleSheet, LogBox} from 'react-native';
import Toast from 'react-native-toast-message';
import {GestureHandlerRootView} from 'react-native-gesture-handler';
import {SafeAreaProvider} from 'react-native-safe-area-context';

import AppContextProvider from './src/context/AppContext';
import MainNavigation from './src/navigation/main';
import NoInternetToast from './src/component/general/noInternet';

// https://github.com/facebook/react-native/issues/33557#issuecomment-1093083115
LogBox.ignoreLogs(["exported from 'deprecated-react-native-prop-types'."]);

const App = () => {
  return (
    <AppContextProvider>
      <GestureHandlerRootView style={styles.gesture}>
        <SafeAreaProvider>
          <MainNavigation />
          <NoInternetToast />
        </SafeAreaProvider>
      </GestureHandlerRootView>
      <Toast />
    </AppContextProvider>
  );
};

const styles = StyleSheet.create({
  gesture: {
    flex: 1
  },
  loadingImage: {
    width: 200,
    height: 200
  }
});

export default App;
