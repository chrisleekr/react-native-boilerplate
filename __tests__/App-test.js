import 'react-native';
import React from 'react';

import {cleanup, render} from '@testing-library/react-native';

import App from '../App';

jest.mock('@react-native-community/netinfo', () => ({
  useNetInfo: () => ({
    isConnected: true
  })
}));

jest.mock('../src/navigation/main', () => props => (
  <main-navigation testID="main-navigation" {...props} />
));

jest.mock('../src/component/general/noInternet', () => props => (
  <no-internet-toast testID="no-internet-toast" {...props} />
));

let rendered;
describe('renders correctly', () => {
  afterEach(cleanup);

  beforeEach(() => {
    rendered = render(<App />);
  });

  it('matches snapshot', async () => {
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('renders main navigation', () => {
    expect(rendered.queryByTestId('main-navigation')).toBeTruthy();
  });

  it('renders no internet toast', () => {
    expect(rendered.queryByTestId('no-internet-toast')).toBeTruthy();
  });
});
