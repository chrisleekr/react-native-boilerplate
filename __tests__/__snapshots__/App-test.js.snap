// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly matches snapshot 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <main-navigation
      testID="main-navigation"
    />
    <no-internet-toast
      testID="no-internet-toast"
    />
  </View>,
  <Toast />,
]
`;
